bt-core/src/main/java/bt/net/ByteChannelReader.java

```diff
@@ -174,11 +174,11 @@ public int read(ByteBuffer buf) throws IOException {
     }
 
     private long getTimeoutMillis() {
-        return timeout.isPresent()? timeout.get().toMillis() : 0;
+        return timeout.map(Duration::toMillis).orElse(0L);
```

Comment:
Boxing/unboxing introduced here

Reply:
Not that I object though, I'm all for readability, if this method is not called too often. Also, somewhat related: there's a long-standing task to get rid of object ints in PieceSelector and classes, that use it (currently, there are many places, where piece indices are passed as Integer instead of int, because selector returns a Stream<Integer> instead of IntStream).